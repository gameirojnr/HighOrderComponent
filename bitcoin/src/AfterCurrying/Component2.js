import React from 'react';

class Component2 extends React.Component {
    
    handleClick = () => {
        this.props.handleTracker('Component2', {})
    }
    
    render() {
        return (
            <div>
                <h1 onClick={this.handleClick}>Component 2</h1>
            </div>
        );
    }
}

export default Component2;