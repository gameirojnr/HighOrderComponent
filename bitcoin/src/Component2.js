import React from 'react';
import API from './API';

class Component2 extends React.Component {    
    
}

export default Component2;