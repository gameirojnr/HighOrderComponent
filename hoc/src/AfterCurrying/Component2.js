import React from 'react';
import PropTypes from 'prop-types';

class Component2 extends React.Component {
    
    handleClick = () => {
        this.props.handleTracker('Component2', {})
    }
    
    render() {
        return (
            <div>
                <h1 onClick={this.handleClick}>Component 2</h1>
            </div>
        );
    }
}

Component2.defaultProps = {
    handleTracker: () => { },
};

Component2.propTypes = {
    handleTracker: PropTypes.func,
};

export default Component2;