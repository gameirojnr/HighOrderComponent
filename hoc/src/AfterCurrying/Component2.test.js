import Component2 from './Component1';
import React from 'react';
import { shallow } from 'enzyme';

describe('<Component2 />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Component2 />);
    expect(wrapper).toMatchSnapshot();
  });

  it('handles click correctly', () => {
    const funcExample = jest.fn();
    const wrapper = shallow(<Component2 handleTracker={funcExample} />);

    wrapper.find('h1').simulate('click');
    expect(funcExample.mock.calls.length).toBe(1);
  });
});