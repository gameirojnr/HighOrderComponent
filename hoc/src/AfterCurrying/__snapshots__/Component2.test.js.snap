// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Component2 /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Component1
    handleTracker={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1
        onClick={[Function]}
>
        Component 1
</h1>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Component 1",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "Component 1",
      "type": "h1",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1
          onClick={[Function]}
>
          Component 1
</h1>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Component 1",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Component 1",
        "type": "h1",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
